src/human_player_module.c

```diff
@@ -83,6 +85,158 @@ bool validate_card(char *entered_value)
+    {
+        //To be added if the format of user input is different
+        Card_t human_card_choice = map_user_input(user_input);
+
+        //Check if the card is from the human player deck
+        if (!is_human_card(&human_card_choice))
+        {
+            printf("!!Warning!! Please select a card from your deck! \n");
+            return RET_INVALID_CARD;
+        }
+
+        //If valid add the card to the discard_pile
+        if (check_is_valid_card(&human_card_choice))
+        {
+            remove_card_from_deck(&(g_players[HUMAN].cards_on_hand), human_card_choice);
+            add_card_at_end(g_discard_pile, human_card_choice);
+            memcpy(&g_card_on_table, &human_card_choice, sizeof(Card_t));
+            end_turn(HUMAN);
+            g_card_requested = false;
+            return RET_SUCCESS;
+        }
+        else
+        {
+            invalid_card_warning();
+            return RET_INVALID_CARD;
+        }
+    }
+    else
+    {
+        invalid_card_warning();
+        return RET_INVALID_CARD;
+    }
+}
+
+/**
+ * @brief human player discards a normal card
+ *
+ * @return ret_type_e: RET_SUCCESS on success;
+ *                     RET_FAILURE on fail
+ */
+ret_type_e human_process_new_card_request(void)
+{
+    if (g_card_requested)
+    {
+        printf("!!Warning!! You've already drawn a card from the pile. Please discard card or end turn now \n");
+        return RET_FAILURE;
+    }
+    else
+    {
+        return request_card(HUMAN, 1);
+    }
+}
+
+/**
+ * @brief human player discards a action card - Wild
+ *
+ * @return ret_type_e: RET_SUCCESS on success;
+ *                     RET_INVALID_CARD on invalid input
+ */
+ret_type_e human_process_action_wild_card(void)
+{
+    CardColor_e color_changed = 0;
+    char colornum = ' ';
+    Card_t updated_card = { ACTION, WILD };
+
+    printf("Please enter your choice for the color changing. (R/B/G/Y)\n");
+    scanf(" %c", &colornum);
+
+    if (colornum == 'R') {
+        color_changed = RED;
+    }
+    else if (colornum == 'G') {
+        color_changed = GREEN;
+    }
+    else if (colornum == 'B') {
+        color_changed = BLUE;
+    }
+    else if (colornum == 'Y') {
+        color_changed = YELLOW;
+    }
+    else
+    {
+        printf("!!Warning!! Choose a valid color (R/B/G/Y) \n");
+        return RET_INVALID_CARD;
+    }
+
+    //If valid add the card to the discard_pile
+    if (!is_human_card(&updated_card))
+    {
+        printf("!!Warning!! Please select a card from your deck! \n");
+        return RET_INVALID_CARD;
+    }
+
+    remove_card_from_deck(&(g_players[HUMAN].cards_on_hand), updated_card);
+    add_card_at_end(g_discard_pile, updated_card);
+
+    //update g_card_on_table
+    g_card_on_table.color = color_changed;
+    g_card_on_table.name = WILD;
+
+    //End turn
+    end_turn(HUMAN);
+    g_card_requested = false;
+    return RET_SUCCESS;
+}
+
+/**
+ * @brief Check if the player input is a action card discarding request
```

Comment:
Can you change this to - "Check if the card discarded by player is an action card"?

Reply:
Done.